motor/vault/credentials_resolver/cached_vault.go

```diff
@@ -0,0 +1,41 @@
+package credentials_resolver
+
+import (
+	"context"
+
+	"go.mondoo.com/cnquery/motor/vault"
+)
+
+type cachedVault struct {
```

Comment:
The cached vault should move into `vault/cache` package since it can be used independently from the resolver